src/Microsoft.Extensions.Configuration.AzureAppConfiguration/ConfigurationClients/IConfigurationClient.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+//
+
+using Azure;
+using Azure.Data.AppConfiguration;
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Extensions.Configuration.AzureAppConfiguration.ConfigurationClients
+{
+    internal interface IConfigurationClient
```

Comment:
I don't see the need of this interface. We could just use your concrete type `FailOverSupportedConfigurationClient` everywhere directly. I also don't expect an interface named like this live in the config provider.

Reply:
We need this interface for - 
1. It dictates the methods we are using from `ConfigurationClient`.
2. If we decide to short-circuit `ConfigurationClient` when using `connectionString` as per [this comment](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/292#discussion_r839141438) then this interface will serve a contract we use for multiple types of clients.
3. We can mock our unit tests better when we have an interface definition. Mocking a concrete type puts many limitations.

Reply:
For 1 and 2, a concrete type can achieve the same.

For 3, the tests were mocked for `ConfigurationClient`, which is a concrete type. Why is there a problem with a different concrete type?

Reply:
tests are mocked for `ConfigurationClient` because the methods in `ConfigurationClient` are virtual methods. I can't see why we would want to make methods in `FailOverSupportedConfigurationClient` virtual if we don't expect that type to be derived further.

Reply:
Why not? Testability is a good reason to make them virtual. I don't want to go that far, but as you mentioned, you may even want to have another client with a different failover strategy.